src/data/rat/floor.lean

```diff
@@ -43,12 +44,15 @@ theorem le_floor {z : ℤ} : ∀ {r : ℚ}, z ≤ floor r ↔ (z : ℚ) ≤ r
   exact int.le_div_iff_mul_le h'
 end
 
-theorem floor_lt {r : ℚ} {z : ℤ} : floor r < z ↔ r < z :=
-lt_iff_lt_of_le_iff_le le_floor
-
 theorem floor_le (r : ℚ) : (floor r : ℚ) ≤ r :=
 le_floor.1 (le_refl _)
 
+instance : floor_ring ℚ :=
+{ floor := rat.floor, le_floor := @rat.le_floor }
+
+theorem floor_lt {r : ℚ} {z : ℤ} : floor r < z ↔ r < z :=
```

Comment:
Please make this `protected theorem` (that way, if you `open rat`, you still have to write `rat.floor_lt`, and `floor_lt` will not be overloaded).

Reply:
> This is great! The construction should indeed be axiomatized generally, before being defined in a particular instance (`rat`).
> 
> Some comments
> 
>     * If you feel like changing it, it would be nice to use `⌊x⌋` instead of `rat.floor` everywhere (in particular in the file `rat/floor`): we want to use the general instance.

I just realised that all the lemmas in `rat.floor` are just duplicates of the lemmas in `algebra.floor`. I removed them all and also moved `nat_ceil` to `algebra.floor`.